Update git-clone task and document changes in behaviour (#427)
tektoncd/catalog: task/git-clone/0.1/README.md

## Usage

> If `revision` is not provided in the params in the taskrun

Reviewer: (nit)
```suggestion
If the `revision` is not provided in the param of the taskrun
```

> then it will take the branch that is specified `default`

Reviewer: ```suggestion
then it will auto-detect the branch as specified by the `default`
```

> in the respective git repository and does the required job.

Reviewer: ```suggestion
in the respective git repository/
```

Author: @chmouel Thanks for suggestions. 

> Perhaps update the sample too, to not use master

will update the `samples` wherever `git-clone` is used.
